### PR TITLE
[Xharness]Enable nullability and update C# to new code patterns.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Cleaning up xharness before new features are added.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs [hunk 1]
```diff
@@ -14,11 +16,11 @@ class AggregatedRunSimulatorTask : AppleTestTask
 		public IEnumerable<RunSimulatorTask> Tasks;
 
 		// Due to parallelization this isn't the same as the sum of the duration for all the build tasks.
-		Stopwatch build_timer = new Stopwatch ();
-		public TimeSpan BuildDuration { get { return build_timer.Elapsed; } }
+		readonly Stopwatch buildTimer = new();
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs:19]:
```suggestion
		readonly Stopwatch buildTimer = new ();
```

File: tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs [hunk 2]
```diff
@@ -14,11 +16,11 @@ class AggregatedRunSimulatorTask : AppleTestTask
...
+		public TimeSpan BuildDuration { get { return buildTimer.Elapsed; } }
 
-		Stopwatch run_timer = new Stopwatch ();
-		public TimeSpan RunDuration { get { return run_timer.Elapsed; } }
+		readonly Stopwatch runTimer = new();
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs:22]:
```suggestion
		readonly Stopwatch runTimer = new ();
```

File: tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs [hunk 3]
```diff
@@ -68,7 +70,7 @@ protected override async Task ExecuteAsync ()
 					ExecutionResult = TestExecutingResult.DeviceNotFound;
 					return;
 				}
-				Jenkins.MainLog.WriteLine ("Selected simulator: {0}", devices.Count() > 0 ? devices.First().Name : "none");
+				Jenkins.MainLog.WriteLine ("Selected simulator: {0}", devices.Any() ? devices.First().Name : "none");
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs:73]:
```suggestion
				Jenkins.MainLog.WriteLine ("Selected simulator: {0}", devices.Any () ? devices.First().Name : "none");
```

**Review by rolfbjarne — APPROVED** [review]

File: tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs [hunk 4]
```diff
@@ -94,7 +96,7 @@ protected override async Task ExecuteAsync ()
 				if (device != null)
```
Comment (tj-devel709) [inline tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs:96]:
```suggestion
				if (device is not null)
```
Might as well throw these in there :)

**Pull request merged**